test/built-ins/Promise/prototype/finally/this-value-non-object.js

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2017 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Jordan Harband
+description: >
+  Promise.prototype.finally called with a non-object-coercible `this` value
+esid: sec-promise.prototype.finally
+features: [Promise.prototype.finally, Function.prototype.call]
+---*/
+
+assert.throws(TypeError, function() {
+  Promise.prototype.finally.call(undefined);
+}, 'undefined');
```

Comment:
I'm not sure what value this adds; it we use `.call` then the function has no more link to `p`, so `p.then` couldn't be reached even if i wanted it to.